### PR TITLE
security: run as non-root user

### PR DESCRIPTION
**Pull request by zhangzujian**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Features
- Bug fixes
- Docs
- Tests

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

File: .github/workflows/build-x86-image.yaml [hunk 1]
```diff
@@ -528,8 +528,12 @@ jobs:
       - name: Load image
         run: docker load --input kube-ovn.tar
 
-      - name: Export debug image tag
-        run: echo "DEBUG_TAG='$(cat VERSION)-debug'" >> "$GITHUB_ENV"
+      - name: Set environment variables
+        run: |
+          if [ $(($RANDOM%2)) -ne 0 ]; then
```
Comment (oilbeater) [inline .github/workflows/build-x86-image.yaml:533]: what does the RANDOM mean here?

Comment (zhangzujian) [reply .github/workflows/build-x86-image.yaml:533]: A random switch to enable valgrind memleak check.

Comment (zhangzujian) [reply .github/workflows/build-x86-image.yaml:533]: Valgrind memleack check cannot be enabled if ovs/ovn is running as a non-root user. But we want to test both running as non-root and memleak check.

Comment (oilbeater) [reply .github/workflows/build-x86-image.yaml:533]: we need a comment to clarify this logical

Comment (zhangzujian) [reply .github/workflows/build-x86-image.yaml:533]: done

**Review by oilbeater — APPROVED** [review]

**Review by oilbeater — APPROVED** [review]

**Pull request merged**